bucketdownload/activity.go

```diff
@@ -37,13 +47,20 @@ func (a *Activity) Execute(ctx context.Context, params *Params) (*Result, error)
 	h := temporal.StartAutoHeartbeat(ctx)
 	defer h.Stop()
 
-	dirPerm := fs.FileMode(0o700)
-	if params.DirPerm != 0 {
-		dirPerm = params.DirPerm
-	}
-
-	if err := os.MkdirAll(params.DirPath, dirPerm); err != nil {
-		return nil, fmt.Errorf("bucketdownload: create directory: %w", err)
+	if params.DirPath != "" {
+		dirPerm := fs.FileMode(0o700)
+		if params.DirPerm != 0 {
+			dirPerm = params.DirPerm
+		}
+		if err := os.MkdirAll(params.DirPath, dirPerm); err != nil {
+			return nil, fmt.Errorf("bucketdownload: create directory: %w", err)
+		}
+	} else {
+		destDir, err := os.MkdirTemp("", "bucketdownload")
+		if err != nil {
+			return nil, fmt.Errorf("bucketdownload: create temp directory: %w", err)
+		}
+		params.DirPath = destDir
```

Comment:
I think it's better not to mutate `params`, I'd set a local variable for the destination dir instead.